pkg/api/meta/restmapper.go

```diff
@@ -269,8 +499,8 @@ func (m *DefaultRESTMapper) AliasesForResource(alias string) ([]string, bool) {
 	return nil, false
 }
 
-// ResourceIsValid takes a string (kind) and checks if it's a valid resource
-func (m *DefaultRESTMapper) ResourceIsValid(resource string) bool {
+// ResourceIsValid takes a partial resource and checks if it's valid
```

Comment:
> Isn't the resource fully-qualified?

One very most common usage of the `RESTMapper` is from the CLI.  When using the CLI, people won't usually fully qualify what they want.  They'll say, `kubectl get foos`, but they probably won't include API group and version.  If that information is enough to find an unambiguous match, we should allow it.  If its not, we should provide a way for them to give us more information to find an unambiguous match.

This gets the internal API ready to support that kind of a match and (for now) keeps us exactly compatible with what we had before.